examples/README.md

```diff
@@ -0,0 +1,13 @@
+# regexp-tree: Examples
+
+Welcome to the `regexp-tree` examples.
+
+Here you'll find ways people have used regexp-tree for fun and profit.
+
+Each example has:
+- A command-line interface
+- A corresponding `lib/` entry for ease of adoption
+
+| File                   | Author                                               | Description       |
+|------------------------|------------------------------------------------------|-------------------|
+| `count-star-height.js` | Jamie Davis [@davisjam](https://github.com/davisjam) | Correct implementation of the well-known and buggy [safe-regex](https://github.com/substack/safe-regex) project. |
```

Comment:
OK, here what I though.

This actually might be a good separate project by itself. Maybe you should create a separate repo for this at all instead of having it in the main regexp-tree repo? With this you fully preserve your own copyright.

Since the example is specific, it might be less practical as specifically an example for a quick start to use the tool. However, I can see it as a good example of already using the tool in other good projects like your research project.

Reply:
OK, so there are really two discussions here:

1. Whether you're interested in `examples/` (and if so, what kind?)
2. Copyright.

I'm certainly not trying to start a fight about this, I just want to make sure we're on the same page.

# To `examples/` or not to `examples/`

> This actually might be a good separate project by itself.

Once the paper is accepted, I will be publishing my research repo to GitHub. The repo includes  this use of `regexp-tree` and a few others, plus a bunch of other unrelated code.

However, I'm loath to just dump my code on the internet and forget about it. I thought this would be a good place to put the `regexp-tree`-specific scripts as a way of giving back to the project.

We could link to a separate project with examples in the README, but then you end up with a link for every person who wants to share examples. Centralization seems better to me in this case.

> Since the example is specific

Of course it's specific :-). The entire `regexp-tree` project is pretty niche. Awesome, but niche.

> it might be less practical as specifically an example for a quick start to use the tool.

There are already many "quick-start" examples in the main README that demonstrate how to use individual features. I view the purpose of `examples/` as to show more complex uses of `regexp-tree` so people can get a sense of how they might want to use it themselves.

Did you have a different kind of code in mind for `examples/`?

# Copyright

> With this you fully preserve your own copyright.

If you want the examples it seems like we can address this with appropriate amendments to the copyright line in a few files.

Reply:
@davisjam, yeah, that's correct, I'd like regexp-tree to maintain base core functionality, don't overload it with "concrete sub-projects" (like your research project), and avoid explicitly promoting/depending on other projects with direct links in README, etc.

All the contributors to this specific repo are explicitly mentioned in the https://github.com/DmitrySoshnikov/regexp-tree/graphs/contributors, with their explicit commits, etc, so their "copyright" is automatically "preserved" (assuming the MIT License of this project).

With this, the `examples\` directory I think should be completely independent from other projects (even as an "example dependency"), less focus on individual names, pages, etc, and more focus on actual examples which can help start using regexp-tree for some needs. These examples, as it seems to me, can be in general still very simple, but taking some smaller "real-world" issues.

For the specific research project, I really think it'll be better to open a separate repo. This will be much better for it, instead of putting it into some generic repo (which regexp-tree is).